connector/google/google.go

```diff
@@ -257,12 +258,15 @@ func (c *googleConnector) getGroups(email string, fetchTransitiveGroupMembership
 		}
 
 		for _, group := range groupsList.Groups {
-			// TODO (joelspeed): Make desired group key configurable
-			userGroups = append(userGroups, group.Email)
+			if _, exists := (*checkedGroups)[group.Email]; !exists {
```

Comment:
Yeah, I would like to add a test. Would I need to setup a mock for that? Is there an example?

Locally I have logs which demonstrate this. I could add that to the code instead.

Reply:
Oh, there are no tests for Google connector at all...

You can use [this code](https://github.com/dexidp/dex/blob/master/connector/bitbucketcloud/bitbucketcloud_test.go) as an example to write a simple test.

Reply:
The example looked easy enough in that you could send to whatever URL you want but the google connector is using a library and so I couldn't figure out how to do it in a similar fashion.

I attempted both interfaces and embedded structs but couldn't get around the issue of `getGroups` always calling its version of `getGroupsList`. In the end I settled on `getGroups` taking a function as a parameter and that worked.

Hopefully that is okay, because otherwise I fear I might need a lot of hand holding to get something working (not a developer by trade.)

Reply:
@snuggie12, I've implemented some unit testing for your PR on my fork: https://github.com/ichbinfrog/dex/blob/reduce-admin-api-calls/connector/google/google_test.go. 

Feel free to pull / take inspiration from the test file and reuse it since it also adds the required instrumentation for the amount of apiCalls.

TLDR: the basic gist of mocking google APIs is to inject a few client options which will force it to point to a local http server whose behavior you can control:
```go
conn.adminSrv, err = admin.NewService(
   context.Background(), 
   option.WithoutAuthentication(), 
   option.WithEndpoint(ts.URL),
)
```

Reply:
> @snuggie12, I've implemented some unit testing for your PR on my fork: https://github.com/ichbinfrog/dex/blob/reduce-admin-api-calls/connector/google/google_test.go.
> 
> Feel free to pull / take inspiration from the test file and reuse it since it also adds the required instrumentation for the amount of apiCalls.
> 
> TLDR: the basic gist of mocking google APIs is to inject a few client options which will force it to point to a local http server whose behavior you can control:
> 
> ```go
> conn.adminSrv, err = admin.NewService(
>    context.Background(), 
>    option.WithoutAuthentication(), 
>    option.WithEndpoint(ts.URL),
> )
> ```

@ichbinfrog thanks for this. I've been busy moving where I live but I'll def take a look soon.